app/common/update.py

```diff
@@ -231,3 +258,91 @@ def merge_local_db():
+        
+        if os.path.exists(default_path):
+            update_user_config('config', 'installdirectory', default_path)
+        else:
+            # Doesn't exist. Let's prompt the user for it
+            warning_message(
+                title="[dqxclarity] Couldn't Find Directory",
+                message="Could not find DQX directory. Please select\nthe Data folder located in your\nDQX install location."
+            )    
+            
+            # Prompt the user to select their DQX data directory, and check
+            # if the dat0 file exists in this path. If it does, it *should* be legit.
+            # Keep prompting the user until the path is valid
+            while True:
+                dqx_path = askdirectory()
+                dat0_path = dqx_path + "/" + dat0_file
+                
+                if os.path.isfile(dat0_path)
+                    update_user_config('config', 'installdirectory', dqx_path)
+                    break
+                else:
+                    warning_message(
+                        title="[dqxclarity] Invalid Directory",
+                        message="The path you provided is not a valid DQX path.\nPlease select the Data folder located\nin your DQX install location."
+                    )
+    
+    # We have an installation path, so let's check it
+    
+    if config["config"]["installdirectory"]:
+        
+        dqx_path = config["config"]["installdirectory"]
+        dat0_path = dqx_path + "/" + dat0_file
+        
+        # If the dat0 file exists here, it should be legit
+        # and we can move on
+        
+        if os.path.isfile(dat0_path):
+            # First, let's check if an idx file backup already exists
+            idx_path = dqx_path + "/" + idx_file + ".bak"
+            
+            # If it doesn't, let's back it up first with a copy
+            if not os.path.isfile(idx_path):
+                shutil.copy(dqx_path + "/" + idx_file, dqx_path + "/" + idx_file + ".bak")
+            
+            # Now let's download the data files
+            try:
+                logger.info("Downloading DAT1 and IDX files.")
+                dat_request = requests.get(GITHUB_CLARITY_DAT1_URL, timeout=15)
+                idx_request = requests.get(GITHUB_CLARITY_IDX_URL, timeout=15)
+                # Make sure both requests are good before we write the files
+                if dat_request.status_code == 200 and idx_request.status_code == 200:
+                    with open(dqx_path + "\data00000000.win32.dat1", "w+b") as f:
+                        f.write(dat_request.content)
+                    with open(dqx_path + "\data00000000.win32.idx", "w+b") as f:
+                        f.write(idx_request.content)
+                    logger.success("Data files downloaded.")
+                else:
+                    logger.error(f"Failed to download data files.")
+                    input("Press ENTER to exit.")

```

Comment:
Do we want to kill the program if it doesn't update? Or should we output an error message alerting the user that it failed and keep running? What do you think?